cloudinit/sources/DataSourceAzure.py

```diff
@@ -661,6 +661,15 @@ def crawl_metadata(self):
             # fetch metadata again as it has changed after reprovisioning
             imds_md = self.get_metadata_from_imds(report_failure=True)
 
+            # validate imds pps metadata
+            imds_ppstype = self._ppstype_from_imds(imds_md)
+            if imds_ppstype != PPSType.NONE.value:
```

Comment:
imds_ppstype can be None if we fallback to the old API (and we shouldn't report failure).
```suggestion
            if imds_ppstype is not None and imds_ppstype != PPSType.NONE.value:
```